src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -154,41 +169,119 @@ def bsp_resolve_field_factory(
 @rule
 async def bsp_resolve_scala_metadata(
     request: ScalaBSPBuildTargetsMetadataRequest,
+    bash: BashBinary,
     jvm: JvmSubsystem,
     scala: ScalaSubsystem,
     build_root: BuildRoot,
 ) -> BSPBuildTargetsMetadataResult:
     resolves = {fs.resolve.normalized_value(jvm) for fs in request.field_sets}
+    jdk_versions = {fs.jdk for fs in request.field_sets}
     if len(resolves) > 1:
         raise ValueError(
             "Cannot provide Scala metadata for multiple resolves. Please set the "
             "`resolve = jvm:$resolve` field in your `[experimental-bsp].groups_config_files` to "
             "select the relevant resolve to use."
         )
-    resolve = list(resolves)[0]
-    scala_version = scala.version_for_resolve(resolve)
+    (resolve,) = resolves
 
+    scala_version = scala.version_for_resolve(resolve)
     scala_runtime = await Get(
         MaterializeScalaRuntimeJarsResult, MaterializeScalaRuntimeJarsRequest(scala_version)
     )
 
+    #
+    # Extract the JDK paths from an lawful-evil process so we can supply it to the IDE
+    #
+
+    # The maximum JDK version will be compatible with all the specified targets
+    jdk_requests = [JdkRequest.from_field(version) for version in jdk_versions]
+    jdk_request = max(jdk_requests, key=_jdk_request_sort_key(jvm))
+
+    jdk, readlink, = await MultiGet(
+        Get(JdkEnvironment, JdkRequest, jdk_request),
+        Get(ReadlinkBinary, ReadlinkBinaryRequest()),
+    )
+
+    cmd = "leak_paths.sh"
+    leak_jdk_sandbox_paths = textwrap.dedent(
+        f"""\
+        # Script to leak JDK cache paths out of Coursier sandbox so that BSP can use them.
+
+        {readlink.path} {jdk.coursier.cache_dir}
+        {jdk.java_home_command}
+        """
+    )
+    leak_sandbox_path_digest = await Get(
+        Digest,
+        CreateDigest(
+            [
+                FileContent(
+                    cmd,
+                    leak_jdk_sandbox_paths.encode("utf-8"),
+                    is_executable=True,
+                ),
+            ]
+        ),
+    )
+
+    leaked_paths = await Get(
+        ProcessResult,
+        Process(
+            [
+                bash.path,
+                cmd,
+            ],
+            input_digest=leak_sandbox_path_digest,
+            immutable_input_digests=jdk.immutable_input_digests,
+            env=jdk.env,
+            use_nailgun=(),
+            description="Leak JDK cache paths for BSP",
```

Comment:
Maybe s/Leak/Compute/? "Leak" is a term specific to the implementation. Also maybe set `level` to DEBUG or TRACE?